mailpoet/lib/Newsletter/Shortcodes/Categories/Link.php

```diff
@@ -48,21 +47,26 @@ public function process(
     bool $wpUserPreview = false
   ): ?string {
     $subscriptionUrlFactory = SubscriptionUrlFactory::getInstance();
-    $queueModel = $this->getQueueModel($queue);
 
     switch ($shortcodeDetails['action']) {
       case 'subscription_unsubscribe_url':
         return self::processUrl(
           $shortcodeDetails['action'],
-          $subscriptionUrlFactory->getConfirmUnsubscribeUrl($wpUserPreview ? null : $subscriber, self::getSendingQueueId($queue)),
+          $subscriptionUrlFactory->getConfirmUnsubscribeUrl(
+            $wpUserPreview ? null : $subscriber,
+            $queue ? $queue->getId() : null
+          ),
           $queue,
           $wpUserPreview
         );
 
       case 'subscription_instant_unsubscribe_url':
         return self::processUrl(
           $shortcodeDetails['action'],
-          $subscriptionUrlFactory->getUnsubscribeUrl($wpUserPreview ? null : $subscriber, self::getSendingQueueId($queue)),
+          $subscriptionUrlFactory->getUnsubscribeUrl(
+            $wpUserPreview ? null : $subscriber,
+            $queue ? $queue->getId() : null
```

Comment:
As you stated more changes would be needed since the default value for `$queue` is `null`, that's why I decided to not try to change alot for this specific case
```  
public function process(
    array $shortcodeDetails,
    NewsletterEntity $newsletter = null,
    SubscriberEntity $subscriber = null,
    SendingQueueEntity $queue = null,
    string $content = '',
    bool $wpUserPreview = false
  ): ?string ```

Reply:
Okay, I'm okay with that.